src/fork_gc.c

```diff
@@ -1006,7 +1006,8 @@ static FGCError FGC_parentHandleNumeric(ForkGC *gc, RedisModuleCtx *rctx) {
   rm_free(fieldName);
 
   //printf("empty %ld, number of ranges %ld\n", rt->emptyLeaves, rt->numRanges);
-  if (rt && rt->emptyLeaves >= rt->numRanges / 2) {
+  if (RSGlobalConfig.forkGCCleanNumericEmptyNodes &&
```

Comment:
Maybe check it after taking the Redis lock? Otherwise you might need it to be volatile no?